lib/install/management/delete.go

```diff
@@ -101,22 +101,25 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *metadata.V
 	rpRef := conf.ComputeResources[len(conf.ComputeResources)-1]
 	ref, err := d.session.Finder.ObjectReference(d.ctx, rpRef)
 	if err != nil {
-		err = errors.Errorf("Failed to get VCH resource pool (%s): %s", rpRef, err)
+		err = errors.Errorf("Failed to get VCH resource pool %q: %s", rpRef, err)
 		return err
 	}
-	_, ok := ref.(*object.ResourcePool)
-	if !ok {
-		log.Errorf("Failed to find resource pool %s, %s", rpRef, err)
-		return err
+	switch ref.(type) {
+	case *object.VirtualApp:
+	case *object.ResourcePool:
+		//		ok
+	default:
+		log.Errorf("Failed to find virtual app or resource pool %q, %s", rpRef, err)
```

Comment:
Looks like this should return an error here too.

Minor change for the format string, we are using colon as a delimiter, so "... %q: %s" instead of comma.